src/frontend/web_application/src/layouts/Page/components/InstallButton/index.jsx

```diff
@@ -1,43 +1,63 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { Trans } from '@lingui/react';
 import { Button, VerticalMenuItem } from '../../../../components';
 import { withNotification } from '../../../../modules/userNotify';
 import { InstallPromptConsumer } from '../../../../modules/pwa';
 
-const InstallButton = ({ notifySuccess }) => (
-  <InstallPromptConsumer
-    render={(defferedPrompt) => {
-      if (!defferedPrompt) {
-        return null;
-      }
-
-      const handleInstall = async () => {
-        defferedPrompt.prompt();
-        const { outcome } = await defferedPrompt.userChoice;
-
-        if (outcome === 'accepted') {
-          notifySuccess({
-            message: (<Trans id="pwa.feedback.install-success">Caliopen has been installed on your device.</Trans>),
-          });
-        }
-      };
-
-      return (
-        <VerticalMenuItem>
-          <Button display="expanded" center={false} onClick={handleInstall}>
-            <Trans id="pwa.action.install">Install</Trans>
-          </Button>
-        </VerticalMenuItem>
-      );
-    }}
-  />
-);
-
-InstallButton.propTypes = {
-  notifySuccess: PropTypes.func.isRequired,
-};
-InstallButton.defaultProps = {
-};
-
-export default withNotification()(InstallButton);
+@withNotification()
+class InstallButton extends Component {
+  static propTypes = {
+    notifySuccess: PropTypes.func.isRequired,
+  };
+  static defaultProps = {
+  };
+  state = {
+    isInstalled: false,
+  };
+
+  handleInstallFactory = ({ defferedPrompt }) => async () => {
+    const { notifySuccess } = this.props;
+
+    defferedPrompt.prompt();
+    const { outcome } = await defferedPrompt.userChoice;
+
+    if (outcome === 'accepted') {
+      notifySuccess({
+        message: (<Trans id="pwa.feedback.install-success">Caliopen has been installed on your device.</Trans>),
+      });
+      this.setState({ isInstalled: true });
```

Comment:
I don't understand the logic here. What if the PWA is already installed ?

Reply:
if so, the defferedPrompt will be undefined because the event catched in InstallPromptConsumer will never be triggered.

agree that might not be crystal clear in some months

Reply:
I've added a comment